neuralpp/symbolic/expression.py

```diff
@@ -203,9 +197,8 @@ def _convert(cls, from_expression: Expression) -> Expression:
                     f"invalid from_expression {from_expression}: {type(from_expression)}"
                 )
 
-
     def get_return_type(self, number_of_arguments: int) -> ExpressionType:
-        if not isinstance(self.type, Callable):
+        if not isinstance(self.type, type(Callable[..., Any])):
```

Comment:
E.g., `isinstance(int, Callable)` is `True` (because we can write `int()`), but `isinstance(int, type(Callable[..., Any]))` is `False`.